### PR TITLE
Fix regressions in web build

### PR DESCRIPTION
**Pull request by kburtram**
This PR contains various changes to enable web-build in a new pipeline and remove web-build from desktop pipelines.

### CODE REVIEW

File: build/gulpfile.reh.js [hunk 1]
```diff
@@ -437,8 +442,43 @@ function packagePkgTask(platform, arch, pkgTarget) {
 			const sourceFolderName = `out-vscode-${type}${dashed(minified)}`;
 			const destinationFolderName = `vscode-${type}${dashed(platform)}${dashed(arch)}`;
 
+			const rollupAngularTask = task.define(`vscode-web-${type}${dashed(platform)}${dashed(arch)}-angular-rollup`, () => {
+				return rollupAngular(REMOTE_FOLDER);
+			});
+			gulp.task(rollupAngularTask);
+
+			const rebuildExtensions = ['big-data-cluster', 'mssql', 'notebook'];
```
Comment (Charles-Gagnon) [inline build/gulpfile.reh.js:450]: Could you add a comment saying why these ones need to be rebuilt?

File: build/lib/rollup.ts [hunk 2]
```diff
@@ -0,0 +1,109 @@
+/*---------------------------------------------------------------------------------------------
+ *  Copyright (c) Microsoft Corporation. All rights reserved.
+ *  Licensed under the Source EULA. See License.txt in the project root for license information.
+ *--------------------------------------------------------------------------------------------*/
+
+import * as fs from 'fs';
+import * as rollup from 'rollup';
+import * as path from 'path';
+
+// getting around stupid import rules
```
Comment (Charles-Gagnon) [inline build/lib/rollup.ts:10]: Let's not use "stupid" here

(also out of curiosity, what was the issue here?)

File: extensions/admin-tool-ext-win/src/test/extension.test.ts [hunk 3]
```diff
@@ -1,13 +0,0 @@
-/*---------------------------------------------------------------------------------------------
- *  Copyright (c) Microsoft Corporation. All rights reserved.
- *  Licensed under the Source EULA. See License.txt in the project root for license information.
- *--------------------------------------------------------------------------------------------*/
-
-import 'mocha';
-import * as vscode from 'vscode';
-
-describe('Extension activate test', () => {
-	it('Extension should activate correctly', async function (): Promise<void> {
```
Comment (Charles-Gagnon) [inline extensions/admin-tool-ext-win/src/test/extension.test.ts:10]: Why was this deleted?

File: extensions/notebook/src/jupyter/serverInstance.ts [hunk 4]
```diff
@@ -209,7 +209,7 @@ export class PerFolderServerInstance implements IServerInstance {
 		}
 		let notebookDirectory = this.getNotebookDirectory();
 		this._token = await utils.getRandomToken();
-		let startCommand = `"${this.options.install.pythonExecutable}" "${this.notebookScriptPath}" --no-browser --no-mathjax --notebook-dir "${notebookDirectory}" --NotebookApp.token=${this._token}`;
+		let startCommand = `"${this.options.install.pythonExecutable}" "${this.notebookScriptPath}" --no-browser --ip=127.0.0.1 --allow-root --no-mathjax --notebook-dir "${notebookDirectory}" --NotebookApp.token=${this._token}`;
```
Comment (Charles-Gagnon) [inline extensions/notebook/src/jupyter/serverInstance.ts:212]: @chlafreniere FYI (although I recall you had already discussed this? Are there any security concerns with doing this?

File: package.json [hunk 5]
```diff
@@ -95,7 +95,7 @@
     "plotly.js-dist-min": "^1.53.0",
     "reflect-metadata": "^0.1.8",
     "rxjs": "5.4.0",
-    "sanitize-html": "^2.3.2",
+    "sanitize-html": "1.19.1",
```
Comment (Charles-Gagnon) [inline package.json:98]: This got bumped due to a security issue so it'd be preferred not to revert this, why is this necessary?

Comment (kburtram) [reply package.json:98]: The new module doesn't contain a UMD bundle and isn't compatible with webmode.

File: extensions/notebook/src/jupyter/serverInstance.ts (hunk 4 above)
Comment (kburtram) [reply extensions/notebook/src/jupyter/serverInstance.ts:212]: yes, I recall discussing on the previous PR.  I'm planning to see if there is a way to check if we're in web mode from the extension.  or we can add a setting to enable.  likely I'll ask the notebook team to address separately.

File: build/lib/rollup.ts (hunk 2 above)
Comment (kburtram) [reply build/lib/rollup.ts:10]: I'm just restoring code that was incorrectly deleted during previous merge.  though I agree that we can remove that word.

File: extensions/admin-tool-ext-win/src/test/extension.test.ts (hunk 3 above)
Comment (kburtram) [reply extensions/admin-tool-ext-win/src/test/extension.test.ts:10]: oops, that was a change that wasn't meant to be pushed.  thanks!

File: extensions/notebook/src/jupyter/serverInstance.ts (hunk 4 above)
Comment (chlafreniere) [reply extensions/notebook/src/jupyter/serverInstance.ts:212]: Thanks @kburtram. https://github.com/microsoft/azuredatastudio/issues/15927 to track.

File: build/azure-pipelines/web/sql-product-build-web.yml [hunk 6]
```diff
@@ -44,59 +32,175 @@ steps:
...
+- script: |
+    set -e
+    yarn gulp vscode-linux-x64-build-deb
+  displayName: Build Deb
+
+- script: |
+    set -e
+    yarn gulp vscode-linux-x64-build-rpm
+  displayName: Build Rpm
+
```
Comment (caohai) [inline build/azure-pipelines/web/sql-product-build-web.yml:153]: Is there any reason why we want to build deb/rpm in web build as well? I thought we had that in normal linux build already.

Comment (kburtram) [reply build/azure-pipelines/web/sql-product-build-web.yml:153]: I setup the build to produce all the linux artifacts.  primary reason is if I want to compare functionality in web vs. desktop I have all the artifacts together.  perhaps we can remove in the future if we want to save build time, storage space, etc.  but I'd like to keep that enabled for time being.

Comment (caohai) [reply build/azure-pipelines/web/sql-product-build-web.yml:153]: Got it, this makes sense!

File: extensions/notebook/src/jupyter/serverInstance.ts (hunk 4 above)
Comment (kburtram) [reply extensions/notebook/src/jupyter/serverInstance.ts:212]: @chlafreniere @Charles-Gagnon I'll spend some time today to see if we have an existing way to determine from the extension host that we are running in webmode, and update as suggested.  Otherwise, we can likely push and follow-up using the issue to address this separately.

File: package.json (hunk 5 above)
Comment (Charles-Gagnon) [reply package.json:98]: That's unfortunate.

@chlafreniere VS Code seems to use [insane](https://github.com/microsoft/azuredatastudio/blob/main/src/vs/base/common/insane/insane.js) for their HTML sanitizing so something we should consider if there's a risk here using the older package.

File: build/azure-pipelines/web/sql-product-build-web.yml (hunk 6 above)
Comment (Charles-Gagnon) [reply build/azure-pipelines/web/sql-product-build-web.yml:153]: Maybe add a comment about that here so others know why in case someone wants to remove it later on

**Review by Charles-Gagnon — APPROVED** [review]
Looks good, but it seems like you still have a build failure running hygiene

**Comment by Charles-Gagnon** [comment]
@kburtram Looks like the cache key is being calculated incorrectly, 

`Cache restored from key: Linux-cacheNodeModules13-`

The output should be a hash value instead. 

A quick fix would be to disable the caching for now - I can investigate fixing this as a follow up.

**Comment by Charles-Gagnon** [comment]
@kburtram Actually, can you try changing https://github.com/microsoft/azuredatastudio/blob/main/.github/workflows/ci.yml#L231 to `sql-computeNodeModulesCacheKey.ts`? I think that should fix it.

**Comment by kburtram** [comment]
I'm going to merge now to unblock related work.  I'll follow-up on the two open issues above separately.
